tests/test_compression.py

```diff
@@ -48,7 +49,88 @@ def test_gzip_round_trip(self, payload, compresslevel):
 
 class TestLZMA(unittest.TestCase):
     # TODO: https://docs.python.org/3/library/lzma.html
-    pass
+    @given(
+        payload=st.binary(),
+        check=st.sampled_from(
+            [lzma.CHECK_NONE, lzma.CHECK_CRC32, lzma.CHECK_CRC64, lzma.CHECK_SHA256]
+        ),
+        compresslevel=st.integers(0, 9),
+    )
+    def test_lzma_round_trip_format_xz(self, payload, check, compresslevel):
+        result = lzma.decompress(
+            lzma.compress(
+                payload, format=lzma.FORMAT_XZ, check=check, preset=compresslevel
+            )
+        )
+        self.assertEqual(payload, result)
+
+    @given(
+        payload=st.binary(), compresslevel=st.integers(0, 9),
+    )
+    def test_lzma_round_trip_format_alone(self, payload, compresslevel):
+        result = lzma.decompress(
+            lzma.compress(payload, format=lzma.FORMAT_ALONE, preset=compresslevel)
+        )
+        self.assertEqual(payload, result)
+
+    @st.composite
+    def filters(self, draw):
+        filter_ids = draw(
+            st.lists(
+                st.sampled_from(
+                    [
+                        lzma.FILTER_DELTA,
+                        lzma.FILTER_X86,
+                        lzma.FILTER_IA64,
+                        lzma.FILTER_ARM,
+                        lzma.FILTER_ARMTHUMB,
+                        lzma.FILTER_POWERPC,
+                        lzma.FILTER_SPARC,
+                    ]
+                ),
+                max_size=3,
+            )
+        )
+        filter_ids.append(lzma.FILTER_LZMA2)
+        # create filters options
+        filters = []
+        for filter in filter_ids:
+            lc = draw(st.integers(0, 4))
+            lp = draw(st.integers(0, 4 - lc))
+            filters.append(
+                {
+                    "id": filter,
+                    "preset": data.draw(st.integers(0, 9)),
+                    "dict_size": data.draw(st.integers(4000, 1.875e8)),
+                    "lc": lc,
+                    "lp": lp,
+                    "mode": data.draw(
+                        st.sampled_from([lzma.MODE_FAST, lzma.MODE_NORMAL])
+                    ),
+                    "mf": data.draw(
+                        st.sampled_from(
+                            [
+                                lzma.MF_HC3,
+                                lzma.MF_HC4,
+                                lzma.MF_BT2,
+                                lzma.MF_BT3,
+                                lzma.MF_BT4,
+                            ]
+                        )
+                    ),
+                    "depth": data.draw(st.integers(min_value=0)),
+                }
+            )
+        return filters
+
+    @given(payload=st.binary(), data=st.data())
+    def test_lzma_round_trip_format_raw(self, payload, data):
+        # create the list of filter ids
+        filters = self.filters(data.draw)
```

Comment:
```suggestion
    @given(payload=st.binary(), filters=lzma_filters())
    def test_lzma_round_trip_format_raw(self, payload, filters):
```

will give us a less complicated failure report, at least.  You'll need to make `lzma_failures()` a standalone function rather than a method for this, but it's not actually using the `self` parameter.

I would then simplify the strategy until it only generates filters which allow the test to pass, and just leave some comments explaining the limits you find.  It's OK if this ends up *way* simpler than at present - any test is better than the no-test status quo!